### ISSUE TITLE
Allow modifying `_native_to_marshmallow`

### ISSUE THREAD

**Issue by MartinAltmayer** [lovasoa/marshmallow_dataclass#71]
This is a follow-up to #16, which was closed after introducing `NewType`. As explained in the following, the `NewType`-solution is not possible in my case, so I'd like to ask you to reconsider adding a way to make `_native_to_marshmallow` mutable. From the implementation it looks pretty simple to replace `_native_to_marshmallow` by a dynamic look-up to Marshmallow's `Schema.TYPE_MAPPING`. Then I could modify the latter and marshmallow_dataclass would pick up the changes.

I would like to "register" a fixed Marshmallow field that should be used whenever a type appears in a dataclass that is converted via marshmallow_dataclass:
```
@dataclass
class Schema:
    color: Color
```
should be equivalent to
```
@dataclass
class Schema:
    color: Color = field(metadata={  ... some lengthy definitions here ... })
```
Unfortunately, I cannot replace "Color" by a `NewType`, because it is defined in a lower-level library. I would like to avoid having two very similar types, one from the library, and another one for use in schemas.

**Comment by MartinAltmayer** [comment]
Another implementation possibility would be to use the `TYPE_MAPPING` of the `base_schema` (if provided) instead of hardcoding the `marshmallow.Schema.TYPE_MAPPING`. Then I could implement a custom subclass of `marshmallow.Schema` which overwrites `TYPE_MAPPING` and pass this as `base_schema`.

**Comment by lovasoa** [comment]
Reusing the `TYPE_MAPPING` from the base schema indeed sounds like a good idea. I would be opened to a pull request implementing this. Would you be ready to work on it ? It shouldn't be a big change.

**Comment by MartinAltmayer** [comment]
Yes, I will try to create a PR for this.

**Comment by lovasoa** [comment]
Great ! Let me know if you need guidance through the code.

**Comment by MartinAltmayer** [comment]
I created PR #72

**Comment by MartinAltmayer** [comment]
Implemented in 7794dd985168ec5d83c47c222e09d00516012628.
